iroh/src/node.rs

```diff
@@ -389,6 +415,17 @@ where
             downloader,
         );
 
+        let gc_task = if let GcPolicy::Interval(gc_period) = self.gc_policy {
+            tracing::info!("Starting GC task with interval {}s", gc_period.as_secs());
+            let db = self.db.clone();
+            let cp = self.collection_parser.clone();
+            let task = rt
+                .local_pool()
```

Comment:
is it a good idea for this to be on the local executor?

Reply:
not sure. My rationale was that the only thing in any way expensive it does is io. For the mark phase the expensive thing is reading the collections into mem and parsing them, where I think the reading into mem is the expensive part.

For the sweep phase the expensive thing is to delete stuff, which depending on the file system might be very fast (modern file systems) or not (windows, sometimes).

I don't think any of this would benefit from being multithreaded in terms of performance. And as far as I understand it this does not mean that one thread of the local pool is permanently blocked, just that this task will be pinned to one thread of the local pool. But the thread will still be able to do other things, e.g. while this thing is waiting for the next gc.

Reply:
I think there are some extreme cases where io can actually benefit from being multithreaded. But typically if you perform multithreaded io you are just making things worse. So unless having a long running task in a pinned thread pool is in some way evil (I would not know why), I think this is the way to go.